packages/docs-page/server/loaders/remote-content.ts

```diff
@@ -0,0 +1,197 @@
+import moize, { Options } from 'moize'
+import semver from 'semver'
+import renderPageMdx from '../../render-page-mdx'
+import {
+  fetchNavData,
+  fetchVersionMetadataList,
+  fetchDocument,
+} from '../../content-api'
+import { normalizeVersion, stripVersionFromPathParams } from '../../util'
+
+import { DEFAULT_PARAM_ID } from '../consts'
+
+import { DataLoader, DataLoaderOpts } from './types'
+import { getPathsFromNavData } from '../get-paths-from-nav-data'
+
+export interface RemoteContentLoaderContext {
+  params: Record<string, string[]> // {} | { page: ["destroy"] }
+  mainBranch?: string // = 'main',
+  remarkPlugins?: $TSFixMe[]
+  scope?: Record<string, $TSFixMe> // optional, I think?
+}
+
+interface RemoteContentLoaderOpts extends DataLoaderOpts {
+  basePath: string
+  enabledVersionedDocs?: boolean
+}
+
+/**
+ * TODO: export this from future content-api client
+ * @see https://app.asana.com/0/1100423001970639/1201071725174928/f
+ */
+interface VersionMetadataItem {
+  product: string
+  ref: string
+  version: string
+  created_at: string
+  display?: string
+  sha: string
+  isLatest?: boolean
+  fullPath: string
+}
+
+interface VersionSelectItem {
+  name: string
+  label: string
+}
+
+const moizeOpts: Options = { isPromise: true, maxSize: Infinity }
+const cachedFetchNavData = moize(fetchNavData, moizeOpts)
+const cachedFetchVersionMetadataList = moize(
+  fetchVersionMetadataList,
+  moizeOpts
+)
+
+/**
+ * Formats a list of version-metadata to,
+ * be passed to `<VersionSelect versions={[...]} />`.
+ * - Sorts by semver, descending
+ */
+export function mapVersionList(
+  list: VersionMetadataItem[]
+): VersionSelectItem[] {
+  const versions = list
+    .sort((a, b) =>
+      semver.compare(semver.coerce(b.version), semver.coerce(a.version))
+    )
+    .map((e) => {
+      const { isLatest, version, display } = e
+
+      const displayValue = display ?? version
+
+      return {
+        name: isLatest ? 'latest' : version,
+        label: isLatest ? `${displayValue} (latest)` : displayValue,
+      }
+    })
+
+  return versions
+}
+
+export default class RemoteContentLoader implements DataLoader {
+  opts: RemoteContentLoaderOpts
+
+  constructor(opts: RemoteContentLoaderOpts) {
+    this.opts = opts
+
+    if (typeof this.opts.enabledVersionedDocs === 'undefined')
+      this.opts.enabledVersionedDocs =
+        process.env.ENABLE_VERSIONED_DOCS?.toString() === 'true'
+
+    if (!this.opts.paramId) this.opts.paramId = DEFAULT_PARAM_ID
+  }
+
+  loadStaticPaths = async (): Promise<$TSFixMe> => {
```

Comment:
...it looks like `ctx` arg may be missing here

Reply:
`ctx` isn't used in this function, so there's no need to declare it for this loader 👍